# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: accessible/atk/nsMaiInterfaceTableCell.cpp
# Commit: 6d0f4821ddcf
# Full Hash: 6d0f4821ddcfca1b59e4db3befc3ec9c12335891
# Author: James Teh <jteh@mozilla.com>
# Date: 2022-03-23 09:49:32
# Regressor Bug: 1735970
# File Overlap Count: 1
# Description:
#   Bug 1735970 part 3: Add TableAccessibleBase and TableCellAccessibleBase. r=morgan
#   
#   These are needed to support RemoteAccessible tables.
#   Stuff specific to LocalAccessible is still in TableAccessible and TableCellAccessible, which now inherit from the new Base classes.
#   
# ==============================================================================

diff -r d82660fb5408 -r 6d0f4821ddcf accessible/atk/nsMaiInterfaceTableCell.cpp
--- a/accessible/atk/nsMaiInterfaceTableCell.cpp	Wed Mar 23 04:00:25 2022 +0000
+++ b/accessible/atk/nsMaiInterfaceTableCell.cpp	Wed Mar 23 04:00:26 2022 +0000
@@ -117,15 +117,15 @@
 
 static GPtrArray* GetColumnHeaderCellsCB(AtkTableCell* aCell) {
   if (AccessibleWrap* accWrap = GetAccessibleWrap(ATK_OBJECT(aCell))) {
-    AutoTArray<LocalAccessible*, 10> headers;
+    AutoTArray<Accessible*, 10> headers;
     accWrap->AsTableCell()->ColHeaderCells(&headers);
     if (headers.IsEmpty()) {
       return nullptr;
     }
 
     GPtrArray* atkHeaders = g_ptr_array_sized_new(headers.Length());
-    for (LocalAccessible* header : headers) {
-      AtkObject* atkHeader = AccessibleWrap::GetAtkObject(header);
+    for (Accessible* header : headers) {
+      AtkObject* atkHeader = AccessibleWrap::GetAtkObject(header->AsLocal());
       g_object_ref(atkHeader);
       g_ptr_array_add(atkHeaders, atkHeader);
     }
@@ -155,15 +155,15 @@
 
 static GPtrArray* GetRowHeaderCellsCB(AtkTableCell* aCell) {
   if (AccessibleWrap* accWrap = GetAccessibleWrap(ATK_OBJECT(aCell))) {
-    AutoTArray<LocalAccessible*, 10> headers;
+    AutoTArray<Accessible*, 10> headers;
     accWrap->AsTableCell()->RowHeaderCells(&headers);
     if (headers.IsEmpty()) {
       return nullptr;
     }
 
     GPtrArray* atkHeaders = g_ptr_array_sized_new(headers.Length());
-    for (LocalAccessible* header : headers) {
-      AtkObject* atkHeader = AccessibleWrap::GetAtkObject(header);
+    for (Accessible* header : headers) {
+      AtkObject* atkHeader = AccessibleWrap::GetAtkObject(header->AsLocal());
       g_object_ref(atkHeader);
       g_ptr_array_add(atkHeaders, atkHeader);
     }